### ISSUE TITLE
Dice Loss function in functional.py file need attention.

### ISSUE THREAD

**Issue by mzg0108** [yakhyo/crackseg#7]
Your dice score/loss function is giving too high score to not so great of predictions.
Here is an example.
![20160330_163725_1921_1_1_2](https://github.com/user-attachments/assets/32230271-cabc-45c9-aec9-5603b49243e5)
![20160330_163725_1921_1_1_2](https://github.com/user-attachments/assets/e56aca4f-e604-4545-bd01-8d4dfc5b7539)
![20160330_163725_1921_1_1_2_pred](https://github.com/user-attachments/assets/1444b99c-54a0-447d-b05e-4308b4fa966a)
If you can see the images, you would see that predicted image is missing a lot of pixels that should have been classified to the major class.
Your dice score was 0.9748 for this while I used **following code** and it produced the score of 0.3734 which seems reasonable.

`def cutsom_dice_score(inputs, targets):
    eps = 1e-6
    inputs = inputs.flatten()
    inputs = (inputs > 0.5)
    targets = targets.flatten()

    intersection = torch.sum(inputs * targets)
    denominator = torch.sum(inputs) + torch.sum(targets)

    # calculate the dice loss
    dice_score = (2.0 * intersection + eps) / (denominator + eps)

    return dice_score`

Can you please comment on the rationality and thought behind your dice score function?

**Comment by yakhyo** [comment]
The Dice loss I implemented in this repo considers both classes (background and crack), that's the reason for having higher score.
